src/lib.rs

```diff
@@ -31,23 +31,26 @@ impl PartialEq<&str> for AnimalName {
     }
 }
 
+fn str_to_animal_name(s: &str) -> AnimalName {
```

Comment:
do you actually have to do this now that from_str is infallible?

Reply:
What's the alternative? 

I could justify an unwrap of FromStr because I know the Err type is Infallible (which is what I did before). But I kind of like this better because it factors out the infallible conversion and thus no audit of the Infallible error that is being unwrapped in From<PublicKey>

Reply:
meh.. the infallible would justify the unwrap so the extra refactor just seemed less compact.